### ISSUE TITLE
BME280 reports dew point null when humidity is 100%

### ISSUE THREAD

**Issue by MaceMoneta** [arendst/Tasmota#9035]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

BME280 reports dew point null when humidity is 100%

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [forum](https://groups.google.com/d/forum/sonoffusers)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 8.4.0
  - [x] Pre-compiled
  - [ ] Self-compiled
    - [ ] IDE / Compiler used: _____
- [x] Flashing tools used: Tasmotizer / OTA
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```
  Configuration output here:
07:26:24 CMD: Backlog Template; Module; GPIO 255
07:26:24 RSL: RESULT = {"NAME":"Generic","GPIO":[255,255,255,255,255,255,255,255,255,255,255,255,255],"FLAG":15,"BASE":18}
07:26:24 RSL: RESULT = {"Module":{"18":"Generic"}}
07:26:24 RSL: RESULT = {"GPIO0":{"6":"I2C SDA"},"GPIO1":{"0":"None"},"GPIO2":{"5":"I2C SCL"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```
  STATUS 0 output here:
07:27:39 CMD: Status 0
07:27:39 RSL: STATUS = {"Status":{"Module":18,"DeviceName":"hvac","FriendlyName":["hvac"],"Topic":"sonoff","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
07:27:39 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"sonoffs","OtaUrl":"http://thehackbox.org/tasmota/release/sonoff.bin","RestartReason":"Software/System restart","Uptime":"0T22:50:06","StartupUTC":"2020-08-04T12:37:33","Sleep":50,"CfgHolder":4617,"BootCount":21,"BCResetTime":"2020-04-17T23:11:11","SaveCount":44,"SaveAddress":"F9000"}}
07:27:39 RSL: STATUS2 = {"StatusFWR":{"Version":"8.4.0(sensors)","BuildDateTime":"2020-07-29T12:08:32","Boot":31,"Core":"2_7_2_1","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"341/699"}}
07:27:39 RSL: STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":3,"LogHost":"135.82.8.141","LogPort":514,"SSId":["moneta",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008101","2805C8000100060000005AFF000000000000","00000040","00006000"]}}
07:27:39 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":621,"Free":380,"Heap":20,"ProgramFlashSize":1024,"FlashSize":8192,"FlashChipId":"1740EF","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDA8787","0415A005","B7FFBFCD","01DA9BC4","64367CC7","00084052"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,14,16,17,20,21,24,29,34","Sensors":"1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,26,28,31,34,37,39,40,42,43,51,52,55,56,58,59,62,64,66,67,74"}}
07:27:39 RSL: STATUS5 = {"StatusNET":{"Hostname":"hvac","IPAddress":"135.82.8.78","Gateway":"135.82.8.240","Subnetmask":"255.255.255.0","DNSServer":"135.82.8.1","Mac":"A0:20:A6:07:04:7D","Webserver":2,"WifiConfig":5,"WifiPower":17.0}}
07:27:39 RSL: STATUS7 = {"StatusTIM":{"UTC":"2020-08-05T11:27:39","Local":"2020-08-05T07:27:39","StartDST":"2020-03-08T02:00:00","EndDST":"2020-11-01T02:00:00","Timezone":99,"Sunrise":"05:58","Sunset":"20:06"}}
07:27:39 RSL: STATUS10 = {"StatusSNS":{"Time":"2020-08-05T07:27:39","BME280":{"Temperature":59.5,"Humidity":88.9,"DewPoint":56.2,"Pressure":1012.8},"PressureUnit":"hPa","TempUnit":"F"}}
07:27:39 RSL: STATUS11 = {"StatusSTS":{"Time":"2020-08-05T07:27:39","Uptime":"0T22:50:06","UptimeSec":82206,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"moneta","BSSId":"78:8A:20:21:81:20","Channel":6,"RSSI":54,"Signal":-73,"LinkCount":1,"Downtime":"0T00:00:07"}}}

```
- [ ] Provide the output of the Console log output when you experience your issue; if applicable:
  _(Please use_ `weblog 4` _for more debug information)_
```
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Raise humidity to 100%

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

I think the dew point should be equal to the temperature when the humidity is 100%

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

![image](https://user-images.githubusercontent.com/43620525/89407779-621ecb80-d6ed-11ea-99c7-5156a3a90f65.png)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by MaceMoneta** [comment]
"A relative humidity of 100% indicates the dew point is equal to the current temperature and that the air is maximally saturated with water."
https://en.wikipedia.org/wiki/Dew_point

**Comment by Staars** [comment]
This line will fail for a humidity of 100 (resulting in x=1):
 https://github.com/arendst/Tasmota/blob/934ca43463e33799ca023453a7b78757f09f7394/tasmota/support_float.ino#L119

**Comment by TrDA-hab** [comment]
For me this means the sensor is broken and needs a replacement soon. After reset, the sensor works normally for a while. Maximum outdoor life 6-9 months. The sensor works indoors for 10-12 months.

This information applies to Chinese sensors with Aliexpress.

![789](https://user-images.githubusercontent.com/56403720/89440468-531f3400-d754-11ea-804b-1e85ce5e5887.jpg)

**Comment by MaceMoneta** [comment]
The sensor isn't broken. The dew point is a calculated value, based on the temperature and relative humidity (both of which are returning correct values). It's a simple calculation error.

**Comment by TrDA-hab** [comment]
**For information only!**
1. Humidity physically cannot be 100%, maximum 99.99%.
2. If your sensor shows 100%, then:
  - the dew point can be calculated correctly.
  - your sensor will be broken soon.
  - reset helps several times.
3. You need to buy a new sensor and make a replacement.

**Comment by MaceMoneta** [comment]
Since the spec sheet indicates the range for relative humidity to be "0...100%", I disagree.

https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BME280-DS002.pdf

**Comment by MaceMoneta** [comment]
Note the spec sheet sample code:

![image](https://user-images.githubusercontent.com/43620525/89448264-55b66500-d725-11ea-88a6-5669e2f10499.png)

**Comment by TrDA-hab** [comment]
It's fine.
Yesterday I replaced two of the six sensors again, and again I have no problem with 100% humidity. And I know 100% that in 6-9 months I will have to change the sensors again. This is my experience, sorry, but Tasmota has no error, this is the quality problem of the sensors.

**Comment by joba-1** [comment]
@TrDA-hab have you even looked?
Or are you so blinded by prejudices about china that you can’t see it?
AFAICS @MaceMoneta has analysed it perfectly:

Humidity 100 leads to log(1) which leads to division by zero in the cited line instead of returning zero. Clearly wrong and completely independent from the sensor.

**Comment by TrDA-hab** [comment]
@joba-1 
> @TrDA-hab have you even looked?
Not. I described my practice. Sorry if I disappointed you.
But you have to understand 100% humidity does not exist, it is absurd. Good luck.

**Comment by joba-1** [comment]
> But you have to understand 100% humidity does not exist, it is absurd.

Me and wikipedia disagree on that. It is not absurd, relative humidity of 100% even has its own name: "dew point". The german (Bosch) sensor seems to be fine :)

**Comment by TrDA-hab** [comment]
I have been looking for a solution to the problem for a very long time and have tried many options. The problem with BME280 "freezing", more details here:
https://github.com/arendst/Tasmota/issues/9035
1. A very large I2C line (1.1m + 1.8m) == unstable power supply (starvation) simply disables the BME280.
I wrote how I solved it here:
https://discordapp.com/channels/479389167382691863/526796261714821151/728640555541856327
2. Very high current at the start of MH-Z19V ~ 0.15A. I added a 1000uF 10V aluminum capacitor in parallel to the power supply.
3. Very high current at start SDS011 ~ 0.2A. I added a 1000uF 10V aluminum capacitor in parallel to the power supply.
This is how the connection diagram looks now - everything works stably for 2 months.
![Multi-sensor 221](https://user-images.githubusercontent.com/56403720/92510190-20190600-f214-11ea-9ce4-141f9747109e.jpg)
![20200908_203218](https://user-images.githubusercontent.com/56403720/92510200-25765080-f214-11ea-896d-94085e5c9b11.jpg)
